src/components/MemoryGraph.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Line } from 'react-chartjs-2';
+import useInterval from '../hooks/use-interval';
+
+/* this line chart graphs the last 10 data points we've seen,
+and updates the lower and upper bound of the range to be equal
+to the min and max of this dataset, respectively. */
+const numData = 10;
+
+const MemoryGraph = ({ next }) => {
+  const [numArray, addToNumArray] = useState([]);
+  const [min, setMin] = useState(0);
+  const [max, setMax] = useState(2500000);
+
+  const setRange = (newArr) => {
+    setMin(Math.min(...newArr));
+    setMax(Math.max(...newArr));
+  };
+
+  useInterval(() => {
+    // const answer = (parseFloat(next) / total) * 100
+    console.log(next);
+    if (numArray.length === numData) {
+      const temp = numArray.filter(((ele, i) => i !== 0));
+      const newArr = temp.concat(next);
+      setRange(newArr);
+      addToNumArray(newArr);
+    } else {
+      const newArr = numArray.concat(next);
+      setRange(newArr);
+      addToNumArray(newArr);
+    }
+  }, 1000);
+
+  console.log(numArray);
+  const data = {
+
+    // i want to come up with better label names
+    // these are also random style attributes which i copied, but the color does look nice
+    labels: ['10', '9', '8', '7', '6', '5', '4', '3', '2', '1'],
```

Comment:
what do you think the label names should be? should we figure these out now?

Reply:
i think its ok like this, theyre just enumerating the last X datapoint we've seen. maybe when youre playing with the graph, you can see if theres something in the library that allows you to add labels to axis, which im sure there is